Lib/test/test_typing.py

```diff
@@ -8287,6 +8368,11 @@ def test_hash_eq(self):
             {Annotated[int, 4, 5], Annotated[int, 4, 5], Annotated[T, 4, 5]},
             {Annotated[int, 4, 5], Annotated[T, 4, 5]}
         )
+        # Unhashable `metadata` raises `TypeError`:
+        self.assertRaises(TypeError, hash, Annotated[int, []])
+        class A:
+            __hash__ = None
+        self.assertRaises(TypeError, hash, Annotated[int, A()])
```

Comment:
I'd prefer to use the contextmanager form here; I find it more readable :)

```suggestion
        with self.assertRaises(TypeError):
            hash(Annotated[int, []])

        class A:
            __hash__ = None
        with self.assertRaises(TypeError):
            hash(Annotated[int, A()]))
```